buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -855,9 +855,11 @@ class BeamModulePlugin implements Plugin<Project> {
       // configurations because they are never required to be shaded or become a
       // dependency of the output.
       def compileOnlyAnnotationDeps = [
-        "com.google.auto.value:auto-value-annotations:1.7",
+        "com.google.auto.value:auto-value-annotations:1.7.2",
         "com.google.auto.service:auto-service-annotations:1.0-rc6",
         "com.google.j2objc:j2objc-annotations:1.3",
+        "org.checkerframework:checker-qual:3.7.0",
```

Comment:
These annotations do end up in the class files so users can benefit from them.

Java spec is that annotations that cannot be resolved on the classpath should be ignored. Unfortunately, the JDK has a bug (don't recall if/when it was fixed) where it actually causes a crash.

See https://github.com/apache/beam/pull/13169 and https://issues.apache.org/jira/browse/BEAM-10632

Reply:
If JDK worked properly, then users who had check qual on their classpath would get the benefit, while other users would not be impacted.

Reply:
So instead, I think the checker-qual needs to be `compile` scope and right here add it to the list of `allowUnusedDeclared`.

Reply:
@kennknowles If I understood your point correctly, you mean to remove checker-qual from compileOnlyAnnotationDeps,

and add here like,
 ```
// Disregard unused but declared (test) compile only dependencies used
      // for common annotation classes used during compilation such as annotation
      // processing or post validation such as spotbugs.
      project.dependencies {
        compileOnlyAnnotationDeps.each { dep ->
          permitUnusedDeclared dep
          permitTestUnusedDeclared dep
        }
        **permitUnusedDeclared "org.checkerframework:checker-qual:$checkerframework_version"**
      }
```

Reply:
Yes, and also add it to `compile` scope deps.

Reply:
Isn't it already in `compile`? https://github.com/apache/beam/blob/7aa7e13c564674a99c99a08b21c97ce92111652a/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L911

Reply:
@kennknowles , But it is already defined in compile scope https://github.com/apache/beam/blob/master/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L911

Reply:
Yea that's what I thought. OK then that part should be fine.

Reply:
corrected as your suggestion @kennknowles